### PR TITLE
Prepare multi OS support

### PR DESCRIPTION
**Pull request by daisukixci**
Support several OS's to run it:
 * Ubuntu
 * Archlinux
 * macOS

### CODE REVIEW

**Review by hush-hush — APPROVED** [review]

File: env.sh [hunk 1]
```diff
@@ -1,15 +1,138 @@
...
+        CLIP="${BIN_PATH}/xclip"
+        CLIP_ARGS="-selection clipboard -i"
+        OPEN="xdg-open"
+        DEPS=(
+            "expect"
+            "gnupg"
+            "pinentry-curses"
+            "git"
+            "yubikey-manager"
+            "xclip"
```
Comment (hush-hush) [inline env.sh:83]: We also need python (not installed by default on a fresh ubuntu VM)

File: gpg.sh [hunk 2]
```diff
@@ -29,27 +33,22 @@ case "$answer" in
     yes|YES|y|Y|Yes)
```
Comment (hush-hush) [inline gpg.sh:33]: In the comment above, let's print the actual deps:

```bash
echo "You need to have ${DEPS[*]} installed on your device."
```

File: env.sh (hunk 1 above)
Comment (daisukixci) [reply env.sh:83]: done

File: gpg.sh (hunk 2 above)
Comment (daisukixci) [reply gpg.sh:33]: done

File: env.sh (hunk 1 above)
Comment (trishankatdatadog) [reply env.sh:84]: Hmm, might be overkill to install Python just to get two random numbers. Should we maybe replace with openssl?

File: git.sh [hunk 3]
```diff
@@ -36,7 +36,7 @@ echo
 
 # Export GPG public key to GitHub.
 echo "Exporting your GPG public key to GitHub."
-$GPG --armor --export "$KEYID" | pbcopy
+$GPG --armor --export "$KEYID" | $CLIP $CLIP_ARGS
```
Comment (trishankatdatadog) [inline git.sh:39]: Why repeat this everywhere? Why not have a `CLIP_CMD=$CLIP $CLIP_ARGS` we could reuse everywhere?

File: env.sh [hunk 4]
```diff
@@ -1,15 +1,139 @@
...
+        GPG_AGENT="${BIN_PATH}/gpg-agent"
+        GPGCONF="${BIN_PATH}/gpgconf"
+        YKMAN="${BIN_PATH}/ykman"
+        CLIP="${BIN_PATH}/xclip"
+        CLIP_ARGS="-selection clipboard -i"
+        OPEN="xdg-open"
+        DEPS=(
+            "expect"
+            "git"
+            "gnupg"
```
Comment (hush-hush) [inline env.sh:80]: we install `gnupg` but the binary if `gpg` so we fail later when we check that everything is in place.

Comment (daisukixci) [reply env.sh:80]: Thanks for finding this, I replace with the gpg package which is an alias of gnupg

File: env.sh (hunk 1 above)
Comment (daisukixci) [reply env.sh:88]: it will be installed by dev onboarding anyway, so your call

File: git.sh (hunk 3 above)
Comment (daisukixci) [reply git.sh:39]: Already enough variable x)

File: env.sh (hunk 4 above)
Comment (daisukixci) [reply env.sh]: done

File: env.sh [hunk 5]
```diff
@@ -1,15 +1,145 @@
...
+        ;;
+    *)
+        OS="not detected"
+        ;;
+esac
+
+echo "OS detected is $OS"
+read -rp "Is it correct ? (y|n)" input
+case $input in
+    yes|y|Yes|YES|YeS|yeS)
```
Comment (lucaspimentel) [inline env.sh:25]: Should this list include capital `Y`?

Comment (daisukixci) [reply env.sh:25]: yes it should, adding it thanks

Comment (daisukixci) [reply env.sh]: done

File: env.sh [hunk 6]
```diff
@@ -1,15 +1,145 @@
...
+        ;;
+    *)
+        OS="not detected"
+        ;;
+esac
+
+echo "OS detected is $OS"
+read -rp "Is it correct ? (y|n)" input
+case $input in
+    Y|yes|y|Yes|YES|YeS|yeS)
```
Comment (ianferguson) [inline env.sh:25]: what about `yES`? -- at this point you almost might as well pipe it through `tr 'A-Z' 'a-z'` and then compare for y and yes only, if you want to support so many variations of yes

File: env.sh [hunk 7]
```diff
@@ -1,15 +1,145 @@
...
+    linux*)
+        OS=$(which_flavour)
+        ;;
+    *)
+        OS="not detected"
+        ;;
+esac
+
+echo "OS detected is $OS"
+read -rp "Is it correct ? (y|n)" input
```
Comment (ianferguson) [inline env.sh:23]: N should be capitalized here as a hint that it is the default option

File: env.sh (hunk 6 above)
Comment (daisukixci) [reply env.sh:25]: done

File: env.sh (hunk 7 above)
Comment (daisukixci) [reply env.sh]: done

File: env.sh [hunk 8]
```diff
@@ -1,15 +1,146 @@
...
+    *)
+        OS="not detected"
+        ;;
+esac
+
+echo "OS detected is $OS"
+read -rp "Is it correct ? (y|N)" input
+input=$(echo "$input" | tr "[:lower:]" "[:upper:]")
+case $input in
+    y|yes)
```
Comment (hush-hush) [inline env.sh:26]: Because of the upper above
```suggestion
    Y|YES)
```

**Review by hush-hush — APPROVED** [review]
Thanks a LOT for this. I was able to run the gpg.sh script in a ubuntu VM (vmware workstation) and it works fine (commits are now signed !).

File: README.md [hunk 9]
```diff
@@ -40,7 +40,10 @@ but that is out of our scope.)
 <s>About 2-3 hours.</s> 15 minutes could save you 15% or more on cybersecurity
 insurance.
 
-You will need macOS, [Homebrew](https://brew.sh/), a password manager, and a
+You will need:
```
Comment (trishankatdatadog) [inline README.md:43]: I recommend just saying:

"You will need macOS + [Homebrew](https://brew.sh/) or Ubuntu or Archlinux, a password manager, and a

**Review by trishankatdatadog — APPROVED** [review]
💯 👏🏽 🚀

**Pull request merged**